docs/monitoring-aws-lambda.asciidoc

```diff
@@ -0,0 +1,69 @@
+
+[[aws-lambda-config-options]]
+== Configuration Options for APM on AWS Lambda
+
+The recommended way of configuring the APM Lambda Extension and the APM Agents on AWS Lambda is through the Lambda function's environment variables. 
+
+The configuration options for the APM Agents are documented in the corresponding language agents:
+
+* {apm-node-ref}/configuration.html[Configuration options - Node.js APM Agent]
+* {apm-py-ref}/configuration.html[Configuration options - Python APM Agent]
+* {apm-java-ref}/configuration.html[Configuration options - Java APM Agent]
+
+The following configuration options are in particular relevant for Elastic's APM on AWS Lambda:
+
+=== `ELASTIC_APM_LAMBDA_APM_SERVER`
+This required config option controls where the Lambda extension will ship data. This should be the URL of the final APM Server destination for your telemetry.
+
+=== `ELASTIC_APM_SECRET_TOKEN` or `ELASTIC_APM_API_KEY`
+One of these needs to be set as the authentication method that the extension uses when sending data to the URL configured via `ELASTIC_APM_LAMBDA_APM_SERVER`.
+
+=== `ELASTIC_APM_SERVICE_NAME`
+The configured name of your application or service.  The APM Agent will use this value when reporting data to the APM Server. If unset, the APM Agent will automatically set the value based on the Lambda function name. Use this config option if you want to group multiple Lambda functions under a single service entity in APM.
+
+=== `ELASTIC_APM_DATA_RECEIVER_TIMEOUT_SECONDS`
+The timeout value, in seconds, for the Lambda Extension's server receiving data from the agent. The _default_ is `15`.
+
+=== `ELASTIC_APM_SEND_STRATEGY`
+Whether to synchronously flush APM agent data from the extension to the APM server at the end of the function invocation.
+The two accepted values are `background` and `syncflush`. The _default_ is `syncflush`.
+
+* The `background` strategy indicates that the extension will not flush when it receives a signal that the function invocation
+has completed. It will instead send any remaining buffered data on the next function invocation. The result is that, if the
+function is not subsequently invoked for that Lambda environment, the buffered data will be lost. However, for lambda functions
+that have a steadily frequent load pattern the extension could delay sending the data to the APM server to the next lambda
+request and do the sending in parallel to the processing of that next request. This potentially would improve both the lambda
+function response time and its throughput.
+* The other value, `syncflush` will synchronously flush all remaining buffered APM agent data to the APM server when the
+extension receives a signal that the function invocation has completed. This strategy blocks the lambda function from receiving
+the next request until the extension has flushed all the data. This has a negative effect on the throughput of the function,
```

Comment:
We also have `ELASTIC_APM_DATA_FORWARDER_TIMEOUT_SECONDS`, `ELASTIC_APM_LOG_LEVEL`, and `ELASTIC_APM_DATA_RECEIVER_SERVER_PORT`

Reply:
@estolfo @jlvoiseux  could you add the descriptions for the missing config options here?

Reply:
I'm trying to remember if we want to actually document `ELASTIC_APM_DATA_RECEIVER_SERVER_PORT` or if we have it for testing. In the meantime, let's not document it.

Reply:
Added missing options in [587124e](https://github.com/elastic/apm-aws-lambda/pull/158/commits/587124e607f66d3419e20446382bd67f0f3b8ff1).